File: src/Shared/FidesOS.Comunicacao/Respostas/OrdemDeServico/AlocarTrabalhador/RespostaAlocacaoTrabalhadorJson.cs
namespace FidesOS.Comunicacao.Respostas.OrdemDeServico.AlocarTrabalhador;

public class RespostaAlocacaoTrabalhadorJson
{
  public Guid Id { get; set; } // O Guid da AlocacaoTrabalhador
  public Guid TrabalhadorIdentificacao { get; set; }
  public long ValorTotal { get; set; }

  // Lista aninhada com os detalhes
  public List<RespostaDetalheProducaoJson> Detalhes { get; set; } = [];
}

File: src/Frontend/fidesos-frontend/src/api/contracts/responses/RespostaAlocacaoTrabalhadorJson.ts

import { RespostaDetalheProducaoJson } from "./RespostaDetalheProducaoJson";

export interface RespostaAlocacaoTrabalhadorJson
{
	id: string;
	trabalhadorIdentificacao: string;
	valorTotal: number;
	detalhes: RespostaDetalheProducaoJson[];
}
